controllers/usersignup/usersignup_controller_test.go

```diff
@@ -4259,6 +4259,110 @@ func TestUserSignupStatusNotReady(t *testing.T) {
 	})
 }
 
+// Test the scenario where the user trying to reactivate but their old space still exists
+func TestUserReactivatingWhileOldSpaceExists(t *testing.T) {
+	// given
+	meta := commonsignup.NewUserSignupObjectMeta("", "john.doe@redhat.com")
+	userSignup := &toolchainv1alpha1.UserSignup{
+		ObjectMeta: meta,
+		Spec: toolchainv1alpha1.UserSignupSpec{
+			Userid:   "UserID123",
+			Username: meta.Name,
+		},
+		Status: toolchainv1alpha1.UserSignupStatus{
+			CompliantUsername: "john-doe",
+		},
+	}
+	mur := murtest.NewMasterUserRecord(t, "john-doe", murtest.MetaNamespace(test.HostOperatorNs))
+	mur.Labels = map[string]string{
+		toolchainv1alpha1.MasterUserRecordOwnerLabelKey: userSignup.Name,
+		toolchainv1alpha1.UserSignupStateLabelKey:       "approved",
+	}
+
+	space := spacetest.NewSpace("john-doe",
+		spacetest.WithSpecTargetCluster("member-1"),
+		spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+		spacetest.WithFinalizer(),
+		spacetest.WithDeletionTimestamp())
+
+	key := test.NamespacedName(test.HostOperatorNs, userSignup.Name)
+
+	t.Run("when reactivating usersignup has space in terminating state", func(t *testing.T) {
+		// given
+		// start with a usersignup that has been just reactivated, and MUR has been created
+		userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey] = "approved"
+		userSignup.Labels["toolchain.dev.openshift.com/approved"] = "true"
+		userSignup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey] = "2" // the user signed up twice
+		userSignup.Status.Conditions = []toolchainv1alpha1.Condition{
+			{
+				Type:   toolchainv1alpha1.UserSignupComplete,
+				Status: v1.ConditionTrue,
+				Reason: "Deactivated",
+			},
+			{
+				Type:   toolchainv1alpha1.UserSignupApproved,
+				Status: v1.ConditionTrue,
+				Reason: "ApprovedAutomatically",
+			},
+			{
+				Type:   toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+				Status: v1.ConditionFalse,
+				Reason: "UserNotInPreDeactivation",
+			},
+			{
+				Type:   toolchainv1alpha1.UserSignupUserDeactivatedNotificationCreated,
+				Status: v1.ConditionFalse,
+				Reason: "UserIsActive",
+			},
+		}
+		ready := NewGetMemberClusters(NewMemberClusterWithTenantRole(t, "member1", v1.ConditionTrue))
+		r, req, _ := prepareReconcile(t, userSignup.Name, ready, userSignup, commonconfig.NewToolchainConfigObjWithReset(t, testconfig.AutomaticApproval().Enabled(true)), baseNSTemplateTier, deactivate30Tier, mur, space)
+		InitializeCounters(t, NewToolchainStatus(
+			WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+				"2,internal": 11, // 11 users signed-up 2 times, including our user above, even though she is not active at the moment
+				"3,internal": 10, // 10 users signed-up 3 times
+			}),
+			WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+				string(metrics.Internal): 21,
+			}),
+		))
```

Comment:
```suggestion
```

minor thing - I don't think the counters are required unless checked at the end of the test

Reply:
yep, thanks for catching that. I was getting some errors before if i did not initialize counters, but it was a side effect of copying it from other test case.
fixed in https://github.com/codeready-toolchain/host-operator/pull/744/commits/10d70b84af0481524da6afd097097ace21e77543

Reply:
No worries and thanks for addressing my comments 👍